airflow/providers/apache/beam/operators/beam.py

```diff
@@ -454,13 +455,22 @@ def execute(self, context: 'Context'):
                             job_class=self.job_class,
                             process_line_callback=process_line_callback,
                         )
-                    self.dataflow_hook.wait_for_done(
-                        job_name=dataflow_job_name,
-                        location=self.dataflow_config.location,
-                        job_id=self.dataflow_job_id,
-                        multiple_jobs=self.dataflow_config.multiple_jobs,
-                        project_id=self.dataflow_config.project_id,
-                    )
+                    if dataflow_job_name and self.dataflow_config.location:
+                        if self.dataflow_config.multiple_jobs:
+                            self.dataflow_hook.wait_for_done(
+                                job_name=dataflow_job_name,
+                                location=self.dataflow_config.location,
+                                job_id=self.dataflow_job_id,
+                                multiple_jobs=self.dataflow_config.multiple_jobs,
```

Comment:
maybe can be re-written better.

Reply:
I think I already fixed that one in #20611 -> rebase after I merge and likely this one will not be needed.

Reply:
You might try to rebase now :)

Reply:
looks like its still needed.

```
airflow/providers/apache/beam/operators/beam.py:463: error: Argument "multiple_jobs" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[bool]";
expected "bool"
                                multiple_jobs=self.dataflow_config.multiple_jobs,
```

Reply:
Ah right. That was another "wait".

Reply:
Easy - default is False if not set.

```
mutltiple_jobs = self.dataflow_config.mutliple_jobs if self.dataflow_config.multiple_jobs else False

....
 
        mutltiple_jobs=multiple_jobs,
```